src/coreComponents/constitutive/solid/DelftEgg.hpp

```diff
@@ -0,0 +1,713 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2019 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2019 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2019 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All right reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ *  @file DelftEgg.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_SOLID_DELFTEGG_HPP
+#define GEOSX_CONSTITUTIVE_SOLID_DELFTEGG_HPP
+
+#include "SolidBase.hpp"
+#include "linearAlgebra/interfaces/BlasLapackLA.hpp"
+#include "LvArray/src/tensorOps.hpp"
+
+namespace geosx
+{
+
+namespace constitutive
+{
+
+/**
+ * @class DelftEggUpdates
+ *
+ * Class to provide material updates that may be
+ * called from a kernel function.
+ */
+class DelftEggUpdates : public SolidBaseUpdates
+{
+public:
+  /**
+   * @brief Constructor
+   * @param[in] bulkModulus The ArrayView holding the bulk modulus data for each element.
+   * @param[in] shearModulus The ArrayView holding the shear modulus data for each element.
+   * @param[in] stress The ArrayView holding the stress data for each quadrature point.
+   */
+  DelftEggUpdates( arrayView1d< real64 const > const & shearModulus,
+                   arrayView2d< real64 const > const & refPressure,
+                   arrayView2d< real64 const > const & refStrainVol,
+                   arrayView1d< real64 const > const & recompressionIndex,
+                   arrayView1d< real64 const > const & virginCompressionIndex,
+                   arrayView1d< real64 const > const & cslSlope,
+                   arrayView1d< real64 const > const & shapeParameter,
+                   arrayView2d< real64 > const & newPreConsolidationPressure,
+                   arrayView2d< real64 > const & oldPreConsolidationPressure,
```

Comment:
Is it possible to define just one preconsolidationPressure parameter then update it with hardening? Do we really need two parameters?